doc/source/compression.rst

```diff
@@ -0,0 +1,154 @@
+necessary metadata in file ``__fragment_metadata.tdb``, such as the
+starting location of each compressed tile and the original tile size
+in the case of variable-length attributes (recall that the original tile
+size has fixed size for fixed-length attributes in *both* dense and
+sparse arrays).
+
+You can also specify a different compressor for the coordinates
+tiles as follows (the default is ZSTD with default compression level):
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      .. code-block:: c++
+
+        schema.set_coords_compressor({TILEDB_LZ4, -1});
+
+To maximize the effect of compression on coordinates, TileDB first
+unzips the coordinates tuples and groups the coordinates along
+the dimensions. For instance, coordinates ``(1,2)``, ``(1,3)``, ``(1,5)``
+will be laid out as ``1 1 1 2 3 5`` prior to passing them into
+the compressor. Placing "similar" values next to each generally leads
+to higher compression ratio.
+
+Recall that there are two data files created for a variable-length
+attribute; one that stores the actual cell values and one that
+stores the *starting offsets* of the cell values in the first file.
+TileDB allows you to even specify a compressor for the offsets
+data tiles (the default is ZSTD with default compression level):
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      .. code-block:: c++
+
+        schema.set_offsets_compressor({TILEDB_BZIP2, -1});
+
+
+Choosing a compressor
+---------------------
+
+TileDB offers a variety of compressors to choose from:
+
+    -  `GZIP <http://www.zlib.net/>`__
+    -  `Zstandard <http://facebook.github.io/zstd/>`__
+    -  `Blosc <http://blosc.org/>`__
+    -  `LZ4 <https://github.com/lz4/lz4>`__
+    -  `RLE <https://en.wikipedia.org/wiki/Run-length_encoding>`__
+    -  `Bzip2 <http://www.bzip.org/>`__
+    -  Double-delta
+
+TileDB implements its own version of **double-delta** compression. It is
+similar to the one presented in `Facebook’s
+Gorilla <http://www.vldb.org/pvldb/vol8/p1816-teller.pdf>`__ system. The
+difference is that TileDB uses a fixed bitsize for all values (in
+contrast to Gorilla’s variable bitsize). This makes the implementation a
+bit simpler, but also allows computing directly on the compressed data
+(which we are exploring in the future).
+
+Note that Blosc comes with multiple compressors; BloscLZ is the default
+one, but it works also in combination with LZ4, LZ4HC, Snappy, Zlib, and
+Zstandard. TileDB allows the user to specify any of these compressors
+with Blosc.
+
+Choosing the right compressor for your application is quite challenging,
+as the effectiveness of a compressor heavily depends on the data being
+compressed. Moreover, each compressor offers a *tradeoff between compression
+ratio and compression/decompression speed*. Here are a couple of
+benchmarks that demonstrate this tradeoff:
+
+    -  `Squash Compression Benchmark <https://quixdb.github.io/squash-benchmark/>`__
```

Comment:
should we reduce the amount of data being linked to, pick a couple compression benchmarks from the raw data and plot the results?

Reply:
I guess we can get to that if we have time.